banzai_floyds/profile.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
```

Comment:
Much of this is refactoring. I've tried to make things more explicit between when we are fitting sigma and when we are fitting the fwhm as we had not been consistent leading to issues. I did also increase the fitting window for fitting the profile width as we explicitly need to get constraints on the background. There is a degeneracy between width and the total flux if you don't estimate the background level.